packages/flutter/lib/src/material/radio_list_tile.dart

```diff
@@ -293,58 +296,87 @@ class RadioListTile<T> extends StatelessWidget {
   ///  * [Feedback] for providing platform-specific feedback to certain actions.
   final bool? enableFeedback;
 
+  @override
+  State<RadioListTile<T>> createState() => _RadioListTileState<T>();
+}
+
+class _RadioListTileState<T> extends State<RadioListTile<T>> with TickerProviderStateMixin, ToggleableStateMixin {
+  void _handleChanged(bool? selected) {
+    if (selected == null) {
+      widget.onChanged!(null);
+      return;
+    }
+    if (selected) {
+      widget.onChanged!(widget.value);
+    }
+  }
+
+  @override
+  ValueChanged<bool?>? get onChanged => widget.onChanged != null ? _handleChanged : null;
+
+  @override
+  bool get tristate => widget.toggleable;
+
+  @override
+  bool? get value => widget.selected;
+
   @override
   Widget build(BuildContext context) {
+    final RadioThemeData radioThemeData = RadioTheme.of(context);
     final Widget control = Radio<T>(
-      value: value,
-      groupValue: groupValue,
-      onChanged: onChanged,
-      toggleable: toggleable,
-      activeColor: activeColor,
+      value: widget.value,
+      groupValue: widget.groupValue,
+      onChanged: widget.onChanged,
+      toggleable: widget.toggleable,
+      activeColor: widget.activeColor,
       materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
-      autofocus: autofocus,
+      autofocus: widget.autofocus,
     );
     Widget? leading, trailing;
-    switch (controlAffinity) {
+    switch (widget.controlAffinity) {
       case ListTileControlAffinity.leading:
       case ListTileControlAffinity.platform:
         leading = control;
-        trailing = secondary;
+        trailing = widget.secondary;
         break;
       case ListTileControlAffinity.trailing:
-        leading = secondary;
+        leading = widget.secondary;
         trailing = control;
         break;
     }
+    final Set<MaterialState> activeStates = states..add(MaterialState.selected);
+    final Color? effectiveActiveColor = widget.activeColor
+      ?? radioThemeData.overlayColor?.resolve(activeStates);
```

Comment:
Presumably a stateless widget could compute the `Set<MaterialState>` states parameter here based on `selected`.